File: LAB3/Source Code/ObjectDetectionAndClarificationUsingClarify/Source/MainActivity.java
package com.example.khushbukolhe.predictions;

import android.graphics.Bitmap;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.content.Intent;
import android.provider.MediaStore;
import java.io.IOException;
import android.net.Uri;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

// Created by Khushbu Kolhe.

public class MainActivity extends AppCompatActivity {
    public static final int PICK_IMAGE_REQUEST = 1;

    ImageView imageView;
    TextView textView;
    Button btnGetGallery;
    Button btnPredict;
    Uri imageFileUri;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        imageView = (ImageView) findViewById(R.id.imageView);
        textView = (TextView) findViewById(R.id.textView);

        btnGetGallery = (Button) findViewById(R.id.btnGetGallery);
        btnPredict = (Button) findViewById(R.id.btnPredict);

        btnGetGallery.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent getIntent = new Intent(Intent.ACTION_GET_CONTENT);
                getIntent.setType("image/*");

                Intent pickIntent = new Intent(Intent.ACTION_PICK, android.provider.MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
                pickIntent.setType("image/*");

                Intent chooserIntent = Intent.createChooser(getIntent, "Select Image");
                chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS, new Intent[] {pickIntent});

                startActivityForResult(chooserIntent, PICK_IMAGE_REQUEST);
            }
        });

        btnPredict.setVisibility(View.INVISIBLE);

        btnPredict.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                navigateToIdentifyObjects();
            }
        });

    }

    private  void navigateToIdentifyObjects() {

        Intent intent = new Intent(this, IdentifyObjects.class);

        Bundle extras = new Bundle();
        extras.putString("imageURI",imageFileUri.toString());
        intent.putExtras(extras);

        startActivity(intent);
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data)
    {
        if (requestCode == PICK_IMAGE_REQUEST && resultCode == RESULT_OK && data != null && data.getData() != null) {
            imageFileUri = data.getData();
            try {
                Bitmap bitmap = MediaStore.Images.Media.getBitmap(getContentResolver(), imageFileUri);
                ImageView myImageView = (ImageView) findViewById(R.id.imageView);
                myImageView.setImageBitmap(bitmap);
                btnPredict.setVisibility(View.VISIBLE);

            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }


}

File: LAB6/Source Code/Tensorflowrestapi-flask/SimpleApp/app/src/main/java/com/example/mohammed_2284/simpleapp/MainActivity.kt
package com.example.mohammed_2284.simpleapp

import android.app.Activity
import android.content.Intent
import android.graphics.Bitmap
import android.net.Uri
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.provider.MediaStore
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import com.example.mohammed_2284.simpleapp.retrofit.APIKindaStuff
import com.google.gson.JsonObject
import kotlinx.android.synthetic.main.activity_main.*
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

import android.util.Base64


import java.io.File
import java.io.FileOutputStream
import java.util.*

class MainActivity : AppCompatActivity() {

    private var ivImage: ImageView? = null
    private var textView: TextView? = null
    private var upflag: Boolean? = false
    private var selectedImage: Uri? = null
    private var bitmap: Bitmap? = null
    internal var imagepath = ""
    internal var fname: String = ""
    internal var file = File("")
    val jsonObj = JsonObject()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        ivImage = findViewById<ImageView>(R.id.ivImage) as ImageView
        textView = findViewById<TextView>(R.id.textView) as TextView


        btnPOST.setOnClickListener {

            saveFile(bitmap, file)

            jsonObj.addProperty("imageBase64", encoder())
            //  POST demo
            APIKindaStuff
                    .service
                    .getVectors(jsonObj)
                    .enqueue(object : Callback<ResponseBody> {
                        override fun onFailure(call: Call<ResponseBody>, t: Throwable) {
                            println("---TTTT :: POST Throwable EXCEPTION:: " + t.message)
                            Toast.makeText(applicationContext, t.message, Toast.LENGTH_SHORT).show()
                        }

                        override fun onResponse(call: Call<ResponseBody>, response: Response<ResponseBody>) {
                            if (response.isSuccessful) {
                                val msg = response.body()?.string()
                                textView?.text = msg
                                println("---TTTT :: POST msg from server :: " + msg)
                                Toast.makeText(applicationContext, msg, Toast.LENGTH_SHORT).show()
                            }
                            else{
                                val msg = response.body()?.string()
                                Toast.makeText(applicationContext, msg, Toast.LENGTH_SHORT).show()
                            }

                        }
                    })
        }

        btnGET.setOnClickListener {
            APIKindaStuff
                    .service
                    .greetUser("Team 5")
                    .enqueue(object : Callback<ResponseBody> {
                        override fun onFailure(call: Call<ResponseBody>, t: Throwable) {
                            println("---TTTT :: GET Throwable EXCEPTION:: " + t.message)
                        }

                        override fun onResponse(call: Call<ResponseBody>, response: Response<ResponseBody>) {
                            if (response.isSuccessful) {
                                //val msg = response.body()?.string()
                                val msg = "We are Team 5"
                                println("---TTTT :: GET msg from server :: " + msg)
                                Toast.makeText(applicationContext, msg, Toast.LENGTH_SHORT).show()
                            }
                        }
                    })
        }

        getPic.setOnClickListener {
            val cameraintent = Intent(
                    MediaStore.ACTION_IMAGE_CAPTURE)
            startActivityForResult(cameraintent, 101)
        }
    }

    fun encoder(): String{
        val bytes = File(imagepath).readBytes()
        val base64 = Base64.encodeToString(bytes, Base64.DEFAULT)
        //val base64 = android.util.Base64.encode(bytes, android.util.Base64.DEFAULT)
        return base64
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {

        try {
            when (requestCode) {
                101 -> if (resultCode == Activity.RESULT_OK) {
                    if (data != null) {
                        selectedImage = data.data // the uri of the image taken
                        if ((data.extras!!.get("data") as Bitmap).toString() == "null") {
                            bitmap = MediaStore.Images.Media.getBitmap(this.contentResolver, selectedImage)
                        } else {
                            bitmap = data.extras!!.get("data") as Bitmap
                        }

                        ivImage?.setVisibility(View.VISIBLE)
                        ivImage?.setImageBitmap(bitmap)

                        //                            Saving image to mobile internal memory for sometime
                        val root = applicationContext.filesDir.toString()
                        val myDir = File("$root/predict")
                        myDir.mkdirs()

                        val generator = Random()
                        var n = 10000
                        n = generator.nextInt(n)

                        //                            Give the file name that u want
                        fname = "null$n.jpg"

                        imagepath = "$root/predict/$fname"
                        file = File(myDir, fname)
                        upflag = true
                    }
                }
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }

        super.onActivityResult(requestCode, resultCode, data)
    }

    private fun saveFile(sourceUri: Bitmap?, destination: File) {

        if (destination.exists()) destination.delete()
        try {
            val out = FileOutputStream(destination)
            sourceUri!!.compress(Bitmap.CompressFormat.JPEG, 100, out)
            out.flush()
            out.close()
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }


}